test/integration/admin/api/objects_controller_test.rb

```diff
@@ -0,0 +1,80 @@
+require 'test_helper'
+
+class Admin::Api::ObjectsControllerTest < ActionDispatch::IntegrationTest
+
+  CONTROLLER = Admin::Api::ObjectsController
+
+  def setup
+    @provider = FactoryBot.create(:provider_account)
+
+    host! @provider.admin_domain
+
+    login_provider @provider
+  end
+
+  def test_status_404
+    get admin_api_objects_status_path(object_type: 'service', object_id: 1)
+    assert_response 404
+  end
+
+  def test_status_200
+    service = FactoryBot.create(:service, account: @provider)
+    CONTROLLER::ServiceObject.any_instance.stubs(:authorize).returns(true)
+    get admin_api_objects_status_path(object_type: 'service', object_id: service.id)
+    assert_response 200
+  end
+
+  def test_status_403
+    get admin_api_objects_status_path(object_type: 'code-injection', object_id: 1)
+    assert_response 403
+  end
+
+  def test_status_service
+    service = FactoryBot.create(:service, account: @provider)
+    CONTROLLER.any_instance.expects(:accessible_services).returns(Service.none)
+    get admin_api_objects_status_path(object_type: 'service', object_id: service.id)
+    assert_response 403
+
+    CONTROLLER.any_instance.expects(:accessible_services).returns(Service.all)
+    get admin_api_objects_status_path(object_type: 'service', object_id: service.id)
+    assert_response 200
+
+    service.destroy!
+    get admin_api_objects_status_path(object_type: 'service', object_id: service.id)
+    assert_response 404
+  end
+
+  def test_status_buyer_account
```

Comment:
In our current implementation in porta, buyer users can only be either destroyed for real or not, but we don't destroy them in background yet. This is does not mean that this endpoint for buyers is bad, it is actually good, but what I mean is that we lack the endpoint for tenants :thinking: and it is the hardest one because when we mark a tenant as scheduled for deletion, we disable its domain, so we should be able to enable the domain only for this endpoint or something like that :thinking: